src/machine/asm/execute.S

```diff
@@ -177,6 +177,40 @@
   CALL_MEMSET; \
   POSTCALL;
 
+#define RANDOM_MEMORY \
+  PREPCALL; \
+  shl $CKB_VM_ASM_MEMORY_FRAME_SHIFTS, TEMP1; \
+  addq $CKB_VM_ASM_ASM_CORE_MACHINE_OFFSET_MEMORY, TEMP1; \
+  addq MACHINE, TEMP1; \
+  movq $CKB_VM_ASM_MEMORY_FRAMESIZE, %rcx; \
+5: \
+  cmp $0, %rcx; \
+  je 6f; \
+  push TEMP1;\
+  push %rcx;\
+  CALL_RAND; \
+  pop %rcx; \
+  pop TEMP1; \
+  mov %al, (TEMP1); \
+  sub $1, %rcx; \
+  add $1, TEMP1; \
+  jmp 5b; \
+6: \
+  POSTCALL;
+
+/*
+ * Make sure TEMP1 stores the frame index.
+ */
+#define INITED_MEMORY \
+  movzbl CKB_VM_ASM_ASM_CORE_MACHINE_OFFSET_CHAOS_MODE(MACHINE), TEMP2d; \
+  cmp $0, TEMP2d; \
```

Comment:
Since we already use the rust compiling conditions, we should be able to remove this branch and determine whether to enable or disable the chaos mode in GCC preprocessor.

Reply:
Preprocessor is a compile-time thing, the other feature I think we can enable, is that a user can choose to enable or disable chaos mode at runtime via APIs. The compiling condition merely affects default behavior here.